interface/main/calendar/index.php

```diff
@@ -37,25 +37,32 @@
 require_once 'includes/pnAPI.php';
 require_once("$srcdir/acl.inc");
 
+
+
 // From Michael Brinson 2006-09-19:
 if (isset($_POST['pc_username'])) {
-    $_SESSION['pc_username'] = $_POST['pc_username'];
+    $pc_username=$_POST['pc_username'];
+    $pc_username[0]=filter_var( $pc_username[0],FILTER_SANITIZE_STRING);
```

Comment:
Really not much of a fan of this kind of thing unless necessary. Isn't there a way to escape $pc_username downstream when it is echoed to html via text() or attr() function?

Reply:
I think since the data from the POST/GET is immediately stored in the $_SESSION it will be best to sanitized it before hand and not in the downstream. Otherwise malicious data might be stored in the  $_SESSION and I think that it is dangerous.
What are your thoughts on this?

Reply:
That's a good point regarding session. But definitely opens door to double escape if not done correctly. Couple questions.
What goals do have in sanitizing this variable?(ie. what are you trying to not allow?)
And would a text() or attr() on the result break things?

Reply:
It is indeed a delicate matter. The general purpose and my point of view of this is 
to prevent as much as possible unwanted data from being stored in the session without encoding the data
for example in the case of  framewidth / FILTER_SANITIZE_NUMBER_FLOAT 
this will make framewidth completely safe 
but in the case of   $pc_username / FILTER_SANITIZE_STRING
it will remove unwanted chars but atrr should be additionally added downstream

Reply:
The thing about these things is they change the data. For example, FILTER_SANITIZE_NUMBER_FLOAT, simply removes non numbers. Seems better in this case to instead do `+ 0` which then basically goes to 0 if something is wrong. Same issue on the FILTER_SANITIZE_STRING, which is simply removing things, and who knows what you end up with. I guess after php's past history of magic_quotes and register_globals, I am just not to trustworthy of using these filters and trusting php to do the right thing.

Reply:
Well that is a matter of a decision since doing "100aaaaa111"+0 resault in 100 not 0 
if you prefer this result on returning 100111  I will change it 
waiting for you opinion on this :)

Reply:
Nice rule of thumb here. When in doubt, lets create a wrapper :)
Would place a function called something like sanitizeNumber() in the library/sanitize.inc.php script(which is already included in globals.php) and can use whatever you wish in it (since easy to change if needed down the road). And then can to a check before returning the sanitized value; if the original value is not equal to the processed value, then use error_log() and then a sanitizeNumber() ERROR to the log with the original value and processed value. This will then show us any potential issues.

Reply:
And regarding the pc_username, recommend doing an exercise and following this variable in codebase to see where it is actually used and how's it's used. I am guessing it will be more secure to attack it there rather than making blind upstream changes.

Reply:
btw, for the above sanitizeNumber(), my vote would be to use the `+ 0` since it seems like that is the route that the majority of the codebase already seems to use when doing this.

Reply:
Good idea :)  I added sanitizeNumber()